### PR TITLE
Update Reviewers for PyTorch Distributed team

### PR DESCRIPTION
**Pull request by XilunWu**
- Reflect PyTorch Distributed team member change on the merge rule
- Added new team members since 2021
- Removed one member no longer on PyTorch Distributed team

### CODE REVIEW

**Review by soumith — APPROVED** [review]

**Comment by wz337** [comment]
Thanks for updating the list!

File: .github/merge_rules.yaml [hunk 1]
```diff
@@ -228,9 +228,14 @@
...
   - H-Huang
-  - d4l3k
   - aazzolini
   - kwen2501
+  - XilunWu
+  - wz337
+  - awgu
+  - fegin
+  - kumpera
+  - yhcharles
```
Comment (fduwjj) [inline .github/merge_rules.yaml:238]: Shall we remove Charlies?

Comment (XilunWu) [reply .github/merge_rules.yaml:238]: I think so.

**Pull request closed without merging**